src/Http/Routing/test/UnitTests/EndpointRoutingMiddlewareTest.cs

```diff
@@ -21,7 +21,30 @@ namespace Microsoft.AspNetCore.Routing
     public class EndpointRoutingMiddlewareTest
     {
         [Fact]
-        public async Task Invoke_OnCall_SetsEndpointFeature()
+        public async Task Invoke_OnException_ResetsEndpoint()
+        {
+            // Arrange
+            var httpContext = CreateHttpContext();
+
+            var middleware = CreateMiddleware(next: context => throw new Exception());
+
+            // Act
+            try
+            {
+                await middleware.Invoke(httpContext);
+            }
+            catch
+            {
+                // Do nothing, we expect the test to throw.
+            }
+
+            // Assert
+            var endpoint = httpContext.GetEndpoint();
+            Assert.Null(endpoint);
```

Comment:
This is a good start, but it would be good to verify you can actually change the path, call middleware again, and get a different result.